components/browser/session/src/test/java/mozilla/components/browser/session/engine/middleware/LastAccessMiddlewareTest.kt

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.session.engine.middleware
+
+import mozilla.components.browser.state.action.BrowserAction
+import mozilla.components.browser.state.action.TabListAction
+import mozilla.components.browser.state.action.TabSessionAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.support.test.argumentCaptor
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.whenever
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+
+class LastAccessMiddlewareTest {
+    lateinit var store: BrowserStore
+    lateinit var context: MiddlewareContext<BrowserState, BrowserAction>
+
+    @Before
+    fun setup() {
+        store = mock()
+        context = mock()
+
+        whenever(context.store).thenReturn(store)
+    }
+
+    @Test
+    fun `UpdateLastAction is dispatched when tab is selected`() {
+        var nextInvoked = false
+        val middleware = LastAccessMiddleware()
+        val captor = argumentCaptor<TabSessionAction.UpdateLastAccessAction>()
+
+        middleware.invoke(context, { nextInvoked = true }, TabListAction.SelectTabAction("123"))
+
+        verify(store).dispatch(captor.capture())
+
+        assertEquals("123", captor.value.tabId)
+        assertTrue(captor.value.lastAccess != 0L)
+        assertTrue(nextInvoked)
+    }
+
+    @Test
+    fun `UpdateLastAction is dispatched when a new tab is selected`() {
+        var nextInvoked = false
+        val middleware = LastAccessMiddleware()
+        val captor = argumentCaptor<TabSessionAction.UpdateLastAccessAction>()
+        val tab = createTab("https://mozilla.org", id = "456")
+
+        middleware.invoke(context, { nextInvoked = true }, TabListAction.AddTabAction(tab, true))
+
+        verify(store).dispatch(captor.capture())
+
+        assertEquals("456", captor.value.tabId)
+        assertTrue(captor.value.lastAccess != 0L)
+        assertTrue(nextInvoked)
```

Comment:
See question above about updating the time before the tab is added. I'd recommend testing with an actual store instance, no need to mock that in test if you want to test a middleware.